### PR TITLE
UCP: Implement passing user memh

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Implement user-defined memory handle hint to avoid rcache lookup

## How
Port #7779

### CODE REVIEW

**Comment by yosefe** [comment]
@brminich @hoopoepg can you pls take a look?

File: src/ucp/tag/rndv.c [hunk 1]
```diff
@@ -1209,7 +1216,18 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_matched, (worker, rreq, rndv_rts_hdr, rts_seq),
...
             }
         }
-        /* put protocol is allowed - register receive buffer memory for rma */
+
+        if (rreq->flags & UCP_REQUEST_FLAG_USER_MEMH) {
+            /* At this point we know the datatype is contig */
+            ucp_request_init_dt_reg_from_memh(rreq,
+                                                ep_config->key.rma_bw_md_map,
+                                                rreq->recv.tag.user_memh,
+                                                &rreq->recv.state.dt.contig);
```
Comment (hoopoepg) [inline src/ucp/tag/rndv.c]: strange alignment

**Review by hoopoepg — APPROVED** [review]
just minor comment

**Review by brminich — APPROVED** [review]

**Review by hoopoepg — APPROVED** [review]

**Pull request merged**